### PR TITLE
[sdk-34] cherry-pick jest-expo-enzyme, jest-expo-puppeteer, jest-expo watch-plugins

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

- publish `jest-expo-enzyme`
- publish `jest-expo-puppeteer`
- bump `jest-expo` with watch-plugins

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -255,8 +255,8 @@ jobs:
           working_directory: ~/expo
```
Comment (ide) [inline .circleci/config.yml]: revert this change. if the cherry-pick pulls in changes that don't make sense on the sdk-xx branch, manually exclude those from the commits. for example, this file references bare-expo which doesn't make sense on the sdk-34 branch.

**Review by ide — APPROVED** [review]

**Comment by EvanBacon** [comment]
in the sdk-35 branch
